### PR TITLE
Add a Rust HealthCheck strategy

### PR DESCRIPTION
**Pull request by Swatinem**
Similar to the python functionality, this just bumps the file timestamp in a regular interval.

I think I wired up the flag correctly through all the layers.

Closes SNS-2464.

### CODE REVIEW

File: rust_snuba/benches/processors.rs [hunk 1]
```diff
@@ -153,6 +153,7 @@ fn create_factory(
         concurrency,
         None,
         true,
+        None,
     );
```
Comment (loewenheim) [inline rust_snuba/benches/processors.rs:157]: I think this `new` function is a candidate for refactoring with a `StrategyFactoryOptions` struct or something (not necessarily in this PR).

File: rust_snuba/rust_arroyo/src/processing/strategies/healthcheck.rs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+};
+use crate::types::Message;
+use crate::utils::metrics::{get_metrics, BoxMetrics};
+
+const TOUCH_INTERVAL: Duration = Duration::from_secs(1);
+
+pub struct HealthCheck<Next> {
+    next_step: Next,
+    path: PathBuf,
+    interval: Duration,
```
Comment (loewenheim) [inline rust_snuba/rust_arroyo/src/processing/strategies/healthcheck.rs:15]: Is this field necessary? It's never anything but `TOUCH_INTERVAL`.

**Review by loewenheim — APPROVED** [review]

File: rust_snuba/rust_arroyo/src/processing/strategies/healthcheck.rs (hunk 2 above)
Comment (Swatinem) [reply rust_snuba/rust_arroyo/src/processing/strategies/healthcheck.rs:15]: maybe we want to make this configurable at some point, but you are right, right now its constant.

File: rust_snuba/rust_arroyo/Cargo.toml [hunk 3]
```diff
@@ -9,6 +9,7 @@ edition = "2021"
 chrono = "0.4.26"
 clap = "2.18.0"
 env_logger = "0.9.0"
+filetime = "0.2.22"
```
[the remark this answers is not in the document]
Comment (Swatinem) [reply rust_snuba/rust_arroyo/Cargo.toml:12]: this is what we also use in symbolicator. But okay, there we are touching files which have content, and we can’t just re-create a fresh file.

**Pull request merged**